HdfsAccessor.go

```diff
@@ -118,13 +118,14 @@ func (this *hdfsAccessorImpl) connectToNameNodeImpl(nnAddr string) (*hdfs.Client
 
 // Opens HDFS file for reading
 func (this *hdfsAccessorImpl) OpenRead(path string) (ReadSeekCloser, error) {
```

Comment:
In our discussion on Friday, I had raised concern over us returning readers which are using the same underlying connection. It seems we will be fine, because looking through colinmarc's hdfs code, it seems that all the requests made to namenode via same client are under a lock to prevent simultaneous access.
I agree with your comment in the discussion that this won't affect our perf in anyway.
I think we should remove our own lock from MetadataClient.